manager/dispatcher/assignments.go

```diff
@@ -245,3 +250,29 @@ func (a *assignmentSet) message() api.AssignmentsMessage {
 
 	return message
 }
+
+// secret populates the secret value from raft store. For external secrets, the value is populated
+// from the secret driver.
+func (a *assignmentSet) secret(readTx store.ReadTx, secretID string) (*api.Secret, error) {
+	secret := store.GetSecret(readTx, secretID)
+	if secret == nil {
```

Comment:
@riyazdf, just to clarify, following @cpuguy83 comments, I consolidate all secret fetching functionality to a single function. 
The flow:
1. Fetch the secret (since the secret is required by the task, return error if not found)
2. If secrets driver is defined, fetch the secret value from the driver (otherwise return the secret value)
Let me know if additional refactoring is needed.